crates/telio-wg/src/wg.rs

```diff
@@ -786,10 +813,17 @@ impl State {
 
                 if let Some(stats) = self.stats.get_mut(key) {
                     match stats.lock().as_mut() {
-                        Ok(s) => s.update(
-                            new.rx_bytes.unwrap_or_default(),
-                            new.tx_bytes.unwrap_or_default(),
-                        ),
+                        Ok(s) => {
+                            s.update(
+                                new.rx_bytes.unwrap_or_default(),
+                                new.tx_bytes.unwrap_or_default(),
+                            );
+
+                            if let (Some(tx_ts), Some(rx_ts)) = (s.get_tx_ts(), s.get_rx_ts()) {
+                                self.network_activity_ts = Some(TxRxTimestampPair { tx_ts, rx_ts });
+                            }
```

Comment:
It's either this, or traversing `self.stats` and retrieving timestamps on each `get_ts()` call. Both should be fine, there shouldn't be too much stats to traverse to be honest, but this is slightly more optimized in terms of micro-optimisation :))

Reply:
Actually I think you're right. Consolidation event will be triggered every second or on some events, same for session keeper's update, meaning to calculate and store it only introduces possibility for a bug to appear. I will inline the code in this function instead.

Reply:
@stalowyjez so the actual reason(I remembered when I went to refactor) was that every entry inside of `stats` is under a mutex(not the hashmap itself but rather the values).

This means every time this is called, a mutex is locked, and in real life this is gonna be called on every consolidation which is not that often to be honest.

However it also means that in case of a bug where batcher would resolve prematurely it will call over and over and over again which would be a problem on itself, however with mutexes it would increase the possible damage significantly due to contention.

For the time being I am preferring a simpler approach so I reworked it as per your suggestion.